lib/index.ts

```diff
@@ -326,3 +328,114 @@ export function main() {
+  firstStatementPath.insertBefore(
+    babel.types.importDeclaration(
+      [babel.types.importNamespaceSpecifier(babel.types.identifier("t"))],
+      babel.types.stringLiteral("ts-interface-checker"),
+    )
+  );
+
+  // Get the user config passed to us by babel-plugin-macros, for use as default options
+  // Note: `config` property is missing in `babelPluginMacros.MacroParams` type definition
+  const defaultOptions = ((params as any).config || {}) as ICompilerOptions
+
+  callPaths.forEach(({parentPath}, callIndex) => {
+    // Determine compiler parameters
+    const getArgValue = getGetArgValue(callIndex, parentPath);
+    const file = path.resolve(filename, "..", getArgValue(0) || path.basename(filename));
+    const options = { ...defaultOptions, ...(getArgValue(1) || {}), format: "js:cjs" };
+
+    // Compile
+    let compiled: string | undefined;
+    try {
+      compiled = Compiler.compile(file, options);
+    } catch (error) {
+      throw macroError(`${error.name}: ${error.message}`)
+    }
+
+    // Get the compiled type suite as AST node
+    const parsed = parse(compiled)!;
+    if (parsed.type !== 'File') throw macroInternalError();
+    if (parsed.program.body[1].type !== 'ExpressionStatement') throw macroInternalError();
+    if (parsed.program.body[1].expression.type !== 'AssignmentExpression') throw macroInternalError();
+    const typeSuiteNode = parsed.program.body[1].expression.right;
+
+    // Build checker suite expression using type suite
+    const checkerSuiteNode = babel.types.callExpression(
+      babel.types.memberExpression(
+        babel.types.identifier("t"),
+        babel.types.identifier("createCheckers"),
+      ),
+      [typeSuiteNode],
+    );
+
+    // Replace call with checker suite expression
+    parentPath.replaceWith(checkerSuiteNode);
+  })
+
+  function parse(code: string) {
+    return babel.parse(code, {configFile: false});
+  }
+  function getGetArgValue (callIndex: number, callExpressionPath: babel.NodePath) {
+    const argPaths = callExpressionPath.get("arguments");
+    if (!Array.isArray(argPaths)) throw macroInternalError()
+    return (argIndex: number): any => {
+      const argPath = argPaths[argIndex];
+      if (!argPath) {
+        return null;
+      }
+      const { confident, value } = argPath.evaluate();
+      if (!confident) {
+        /**
+         * TODO: Could not get following line to work:
+         * const lineSuffix = argPath.node.loc ? ` on line ${argPath.node.loc.start.line}` : ""
+         * Line number displayed is for the intermediary js produced by typescript.
+         * Even with `inputSourceMap: true`, Babel doesn't seem to parse inline sourcemaps in input.
+         * Maybe babel-plugin-macros doesn't support "input -> TS -> babel -> output" pipeline?
+         * Or maybe I'm doing that pipeline wrong?
+         */
+        throw macroError(`Unable to evaluate ${ordinal(argIndex + 1)} argument to ${ordinal(callIndex + 1)} call to makeCheckers()`)
```

Comment:
I think `Unable to evaluate argument N to call N to makeCheckers()` would be perfectly understandable, no? The `ordinal` function is a nicety, but seems an overkill for an error message like this.

Reply:
Yup, that sounds good. 

I was actually just trying to avoid any confusion about whether the numbers start at zero (0) or at one (1). Not a big issue. 

I will have the numbers start at one (1) since it's a human-readable string, and I think that should be understood.

Reply:
Fixed in 95b43eb9f875416bd21ceb3cdf9f8636ecacee52